src/gui/dialogs/addon/manager.cpp

```diff
@@ -738,6 +739,21 @@ void addon_manager::publish_addon(const addon_info& addon)
 		}
 	}
 
+	// the passphrase isn't provided, prompt for it
+	if(cfg["passphrase"].str().empty()) {
+		gui2::dialogs::addon_auth auth;
+		auth.show();
+
+		switch(auth.get_retval()) {
+			case gui2::retval::OK:
+				cfg["passphrase"] = auth.get_password();
```

Comment:
Instead of using a getter, I suggest taking `cfg["password"]` as a reference and setting that directly in post_show.

Reply:
Updated.